### ISSUE TITLE
[4.x]: "Showing your unsaved changes." keeps popping up after not making any changes

### ISSUE THREAD

**Issue by Antimated** [craftcms/cms#11106]
### Description
I have a basic entry with just the default title field and a matrix field (that is limited to 1 item). Whenever I open an entry and just click anywhere on the page, the "Showing your unsaved changes." message pops up. If I discard the changes and click anywhere the same thing happens. If I save the entry and click anywhere, the same thing happens.

### Steps to reproduce

1. Open an entry
2. Click somewhere
3. Message pops up wether or not I changed content

### Expected behavior
The message shouldn't pop up when I don't make any changes

### Actual behavior
The message of unsaved changes constantly pops up after I click anywhere on the page

### Craft CMS version

4.0.0.1

### PHP version

8.1.5

### Operating system and version

macOS Big Sur version 11.4

### Database type and version

MySQL 5.7.34

### Image driver and version

GD 8.1.5

### Installed plugins and versions

- Colour Swatches: 4.0.0
- CP Field Inspect: 1.4.3
- Formie: 2.0.0-beta.5
- Knock Knock: 2.0.0-beta.1
- MatrixMate: 2.0.0
- Navigation: 2.0.0-beta.5
- Position Fieldtype: 4.0.0-beta.3
- Ray: 2.0
- Redactor: 3.0.0
- SEOmatic: 4.0.0-beta.8
- Slug Equals Title: 2.0.0-beta.1
- Super Table: 3.0.0-beta.3
- Typed link field: 2.1.1-rc
- Vite: 4.0.0-beta.4

**Comment by brandonkelly** [comment]
It’s most likely due to a field type that is altering its post data after page load, making it look to Craft like something has changed about it. Can you tell me exactly which field types are in use on the page?

**Comment by Antimated** [comment]
> It’s most likely due to a field type that is altering its post data after page load, making it look to Craft like something has changed about it. Can you tell me exactly which field types are in use on the page?

Yeah sure:

- Native title field
- Matrix (with max 1 block)
    - Image
    - Plain text
    - Plain text
    - Link field (typed link field)
    - Dropdown

**Comment by brandonkelly** [comment]
Assuming “Image” is an Assets field?

Typed Link is the most likely culprit. Can you try duplicating the entry, and removing any blocks that use that field type, and see if it’s still reproducible?

**Comment by Antimated** [comment]
> Assuming “Image” is an Assets field?
> 
> Typed Link is the most likely culprit. Can you try duplicating the entry, and removing any blocks that use that field type, and see if it’s still reproducible?

Yes "Image" is an assets field. Well the matrix (and the default title) is the only field that is displayed on this entry type right now, it's a project I've just started on :/

**Comment by brandonkelly** [comment]
What fields are inside the Matrix field?

**Comment by Antimated** [comment]
> What fields are inside the Matrix field?

This is the full matrix layout:

**Full**
- Image (asset field)
- Title (plain text)
- Subtitle (plain text)
- Link (typed link field)
- Text color (dropdown)

**Half**
- Image (asset field)
- Title (plain text)
- Subtitle (plain text)
- Link (typed link field)
- Background color (dropdown)
- Text color (dropdown)
- Image position (Position)

Note: doesn't matter which block type I use, happens on the both of them. Even if I clear out the matrix so it looks like this it has the same behavior:

<img width="1065" alt="Screenshot 2022-05-06 at 09 42 26" src="https://user-images.githubusercontent.com/2869409/167088927-8ca755a2-1e77-4ecb-b6e3-4bc892f8b88f.png">

**Comment by brandonkelly** [comment]
Strange. Not able to reproduce locally.

Can you try disabling each of your plugins, and see if it still happens for you? If not, then re-enable them one-by-one to see which one is the culprit.

**Comment by Antimated** [comment]
> Strange. Not able to reproduce locally.
> 
> Can you try disabling each of your plugins, and see if it still happens for you? If not, then re-enable them one-by-one to see which one is the culprit.

I just disabled Typed link field as a test and instantly stopped this behaviour. I should probably notify them as I have a few errors using that plugin now :( (integrity violations etc)

**Comment by brandonkelly** [comment]
Hm… if you don’t have any Typed Link fields on the page, I’d be a bit surprised if that was the actual culprit. Maybe try re-enabling that, but disabling all non-field type plugins that could be interacting with the page:

- CP Field Inspect
- Knock Knock
- MatrixMate
- Navigation
- Ray
- SEOmatic
- Slug Equals Title
- Vite

**Comment by Antimated** [comment]
It's weird because I disabled and enabled typed link field (that field type is actually present within my matrix as I said before) but now the behaviour is gone...

**Comment by maxstrebel** [comment]
It guess it's a bug with typed link field. We just removed it from our sites, because it causes a myriad of problems on Craft 4.
